client/src/main/webjar/program-details/program-details.component.js

```diff
@@ -6,32 +6,57 @@
     /**
      * Controller for the healthBamProgramDetails component.
      * @param MapQuery
+     * @param Program
+     * @param $state
      * @param $stateParams
      * @param $q
+     * @param programFormDialogService
      * @param errorHandlingService
      * @param mapConfig
      * @param $log
      * @constructor
      */
     function ProgramDetailsController(
         MapQuery,
+        Program,
+        $state,
         $stateParams,
         $q,
+        programFormDialogService,
         errorHandlingService,
         mapConfig,
         $log
     ) {
         var programDetails = this;
 
         /**
+         * Opens a dialog for editing the Program.
+         * @param $event
+         * @returns {promise} resolved when dialog is closed.
+         */
+        function edit($event) {
+            return programFormDialogService.edit(
+                $event,
+                programDetails.program
+            ).then(
+                function (data) {
+                    $state.reload();
+                    return data;
+                }
+            );
+        }
+
+        /**
          * Handles success loading the MapQuery for the program.
          * @param mapQuery response.
          * @returns mapQuery input or rejected promise on error.
          */
         function handleMapQuerySearchSuccess(mapQuery) {
 
             if (mapQuery.result && angular.isArray(mapQuery.result.programs) && mapQuery.result.programs.length === 1) {
-                programDetails.program = mapQuery.result.programs[0];
+                programDetails.program = new Program(
+                    mapQuery.result.programs[0]
```

Comment:
Maybe it doesn't matter b/c the object is flat, so the prototypical inheritance is enough.